### PR TITLE
Add "cpan" Perl Packages type

### PR DESCRIPTION
**Pull request by giterlizzi**

### CODE REVIEW

**Comment by sjn** [comment]
@pombredanne here's the CPAN spec addition, as we discussed! 😊

**Comment by sjn** [comment]
Is it necessary to mention metacpan at all?

File: PURL-TYPES.rst [hunk 1]
```diff
@@ -168,6 +168,35 @@ conda
 
       pkg:conda/absl-py@0.4.1?build=py36h06a4308_0&channel=main&subdir=linux-64&type=tar.bz2
 
+cpan
+----
+``cpan`` for CPAN Perl packages:
+
+- The default respository is ``https://www.cpan.org/``.
+- To search CPAN it is recommended to use ``https://metacpan.org``.
```
Comment (sjn) [inline PURL-TYPES.rst:176]: I think we safely can remove this line, since the module search web page isn't relevant for how PackageURLs work.

Comment (giterlizzi) [reply PURL-TYPES.rst]: Corrected ;)

**Review by pombredanne — APPROVED** [review]
LGTM! Thank you ++

**Comment by giterlizzi** [comment]
Thank you !!!

**Comment by sjn** [comment]
Would it be meaningful to change the text to use only the terms defined in https://datatracker.ietf.org/doc/html/rfc2119? (e.g. MUST, and MUST NOT, and do not use the term ~~NEVER~~)

**Pull request merged**